domain-models-runtime/src/test/java/org/folio/rest/persist/PostgresClientIT.java

```diff
@@ -2219,54 +2222,377 @@ public void selectDistinctOn(TestContext context) throws IOException {
+      facets, context.asyncAssertSuccess(sr -> {
+        ResultInfo resultInfo = sr.resultInto();
+        context.assertEquals(0, resultInfo.getTotalRecords());
+        context.assertEquals(0, resultInfo.getFacets().size());
+        sr.handler(streamHandler -> objectCount.incrementAndGet());
+        sr.endHandler(x -> {
+          context.assertEquals(0, objectCount.get());
+          async.complete();
+        });
+      }));
+    async.awaitSuccess();
+  }
+
+  @Test
+  public void streamGetWithFacetsError(TestContext context) throws IOException, FieldException {
+    AtomicInteger objectCount = new AtomicInteger();
+    final String tableDefiniton = "id UUID PRIMARY KEY , jsonb JSONB NOT NULL, distinct_test_field TEXT";
+    List<FacetField> badFacets = new ArrayList<FacetField>();
+    badFacets.add(new FacetField("'"));  // bad facet
+    createTableWithPoLines(context, MOCK_POLINES_TABLE, tableDefiniton);
+    CQLWrapper wrapper = new CQLWrapper(new CQL2PgJSON("jsonb"), "edition=First edition");
+    postgresClient.streamGet(MOCK_POLINES_TABLE, Object.class, "jsonb", wrapper, true, null,
+      badFacets, context.asyncAssertFailure());
+  }
 
+  @Test
+  public void streamGetWithSyntaxError(TestContext context) throws IOException, FieldException {
+    final String tableDefiniton = "id UUID PRIMARY KEY , jsonb JSONB NOT NULL, distinct_test_field TEXT";
     createTableWithPoLines(context, MOCK_POLINES_TABLE, tableDefiniton);
-    postgresClient.streamGet(MOCK_POLINES_TABLE, new Object(), "jsonb", null, false, "jsonb->>'edition'",
-        streamHandler -> objectCount.incrementAndGet(), asyncResult -> {
+    CQLWrapper wrapper = new CQLWrapper(new CQL2PgJSON("jsonb"), "edition=");
+    postgresClient.streamGet(MOCK_POLINES_TABLE, Object.class, "jsonb", wrapper, false, null,
+      context.asyncAssertFailure());
+  }
+
+  @Test
+  public void streamGetExceptionInHandler(TestContext context) throws IOException, FieldException {
+    final String tableDefiniton = "id UUID PRIMARY KEY , jsonb JSONB NOT NULL, distinct_test_field TEXT";
+    createTableWithPoLines(context, MOCK_POLINES_TABLE, tableDefiniton);
+    CQLWrapper wrapper = new CQLWrapper(new CQL2PgJSON("jsonb"), "edition=First edition");
+    StringBuilder events = new StringBuilder();
+    Async async = context.async();
+    postgresClient.streamGet(MOCK_POLINES_TABLE, Object.class, "jsonb", wrapper, false, null,
+      context.asyncAssertSuccess(sr -> {
+        sr.handler(streamHandler -> {
+          events.append("[handler]");
+          throw new NullPointerException("null");
+        });
+        sr.endHandler(x -> {
+          events.append("[endHandler]");
+          async.complete();
+        });
+        sr.exceptionHandler(x -> {
+          events.append("[exception]");
+        });
+      }));
+    async.awaitSuccess();
+    context.assertEquals("[handler][exception][handler][exception][handler][exception][endHandler]", events.toString());
```

Comment:
This looks strange to me.
After the exception handler is called I do not expect any other handlers being called, neither the item handler nor the end handler.
If you really want that this may happen it should be clearly documented so that the using modules can take care of it (for example by setting a flag that ignores item handler and end handler invocations).

If the exception handler is the last handler that gets invoked the end handler and the exception handler can be merged into a `Handler<AsyncResult<Void>>`.

Reply:
Agreed. I'll change that, then.